src/App.js

```diff
@@ -1,24 +1,63 @@
-import logo from './logo.svg';
+import react, { useState } from 'react';
+
+import foodsJSON from "./foods.json";
 import './App.css';
+import FoodBox from './components/FoodBox';
+import AddFoodForm from './components/AddFoodForm.jsx';
+import Searchbar from './components/Searchbar';
+
+
 
 function App() {
+  const foodArr = [...foodsJSON]
+  const [foodList, setFoodList] = useState(foodsJSON);
+  const [showFood, setShowFood] = useState(foodsJSON)
+  
+
+  const addFood = (food) => {
+    const foodListCopy = [...foodList, food];
+    setFoodList(foodListCopy)
+  };
+
+  const searchFood = (query) => {
+    const filteredFood = foodsJSON.filter((food)=> 
+    foodsJSON.name.toLowerCase().includes(query.toLowerCase())
+    );
+    setShowFood(filteredFood)
```

Comment:
Your `searchFood` is not working because:
- you're filtering through `foodsJSON` (json file) instead of `foodList` (variable);
- inside the filter you're checking the `.name` of `foodsJSON` (json file) instead of checking the name of every single element in the `foodList` array with `food` (placeholder);
- you're only updating `showFood` with `setShowFood` you need to update `foodList` as well;

```
  const searchFood = (query) => {
    const filteredFood = foodList.filter((food) =>
    food.name.toLowerCase().startsWith(query.toLowerCase()));
    setShowFood(filteredFood);
    setFoodList(filteredFood);
  };
```